### ISSUE TITLE
Feature map question

### ISSUE THREAD

**Issue by cmtzmiwa** [eriklindernoren/PyTorch-YOLOv3#406]
Hi everyone, I would appreciate if you could help me with some questions related to YOLOv3 feature maps. Sorry if these questions are kind of dumb or naive, but I'm just getting started in this topic.

I understand that for a 416x416 image we will end up with a 13x13x255 tensor, i.e., 13x13x (85x3 boxes). The first 4 of these 85 attributes correspond to the bbox coordinates (offsets), the 5th is the objectness score and the remaining 80 are class probabilities.

Now, if I plot the 13x13 map related to the 6th attribute (that is the first class) I obtain the following image:

![Diez1FeatureMap(Persona)](https://user-images.githubusercontent.com/30093539/72330523-46ce6f00-367c-11ea-8a13-ae9b4fffae22.png)

where the original image was:

![body1](https://user-images.githubusercontent.com/30093539/72330555-53eb5e00-367c-11ea-91da-42f2d2385297.jpg)

So, my questions are:

1) Are these 13x13 images, the maps related to each category? (referring to the last 80 attributes). For example, I noticed that the 6th attribute corresponds to the category "person". So, is the 6th 13x13 map (shown in the image) associated with the category "person"?

2) What do the pixels with higher values (yellow ones) refer to? Are they pointing to locations where it is more likely to find a person (in this case)?

3) How are the probabilities for each class calculated? I've read that for YOLOv3 the do not rely on Softmax anymore, but they use a regression. How does this regression work?

I really hope you could help me with these issues. Thank you in advance. P.S. Sorry if my English is not that good, I'm trying to do my best. (Please, let me know if you don't understand something).

**Comment by Flova** [comment]
1. There are 3 anchors for each image that are associated with a rough shape for each cell in the 13x13 yolo grid (other values than 13 are possible for other resolutions). For each anchor, in each cell, there are a number of values that are predicted. A label is associated with a cell if the center of the label is inside the cell and the anchor is selected via the shape of the label. The predicted values include offsets in x and y relative to the cell position and multipliers for the anchor shape to get the width and height. There is also exponential scaling involved but it would be a bit too detailed to discuss here. After the box-related things are the object probability which states if there is an object center in this cell + anchor and the class probabilities (80 for coco).
2. The shown feature map shows the output of the person classification independent of the object probability. It is not a map where a person would be but more a map that describes that if there is something here it is a person. You would need to multiply it with the 5th feature map that includes the object probability to get a map that shows cells with the center of a person in it.
3. Indeed YOLOv3 doesn't use any softmax it just uses the plain output and selects the maximum. This "should" be äquivalent in the end as we are only selecting the maximum and don't really care about scale. Softmax scales all values so the sum is always 1 and we have more of a real probability. Taking the outputs of a network as real probabilities is a dangerous topic as they should only indicate a certainty and not a probability which is meaning full in a mathematical way.

Cheers
